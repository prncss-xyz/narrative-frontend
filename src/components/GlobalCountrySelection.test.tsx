import { fireEvent, render, screen } from "@testing-library/react";
import {
  CountrySelectorContext,
  countryString,
  GlobalCountrySelector,
  sortCountriesByName,
  useGlobalCountryList,
} from "./GlobalCountrySelection";

export const countries = [
  {
    countryCode: "US",
    name: "United States",
    storedData: [
      { datasetId: 1, recordCount: 500 },
      { datasetId: 2, recordCount: 0 },
      { datasetId: 3, recordCount: 2500 },
      { datasetId: 4, recordCount: 750 },
      { datasetId: 5, recordCount: 50 },
      { datasetId: 6, recordCount: 1000 },
    ],
  },
  {
    countryCode: "GB",
    name: "United Kingdom",
    storedData: [
      { datasetId: 1, recordCount: 0 },
      { datasetId: 2, recordCount: 1250 },
      { datasetId: 3, recordCount: 2000 },
      { datasetId: 4, recordCount: 1500 },
      { datasetId: 5, recordCount: 500 },
      { datasetId: 6, recordCount: 500 },
    ],
  },
  {
    countryCode: "CA",
    name: "Canada",
    storedData: [
      { datasetId: 1, recordCount: 1200 },
      { datasetId: 2, recordCount: 799 },
      { datasetId: 3, recordCount: 0 },
      { datasetId: 4, recordCount: 0 },
      { datasetId: 5, recordCount: 0 },
      { datasetId: 6, recordCount: 900 },
    ],
  },
];

function Tester() {
  const [countryCodes] = useGlobalCountryList(countries);
  return (
    <>
      <div>
        {countryCodes.map((countryCode) => (
          <div key={countryCode}>{countryCode}</div>
        ))}
      </div>
      <GlobalCountrySelector count={3} countries={countries} />
    </>
  );
}

describe("GlobalCountrySelection", () => {
  it("should toggle country when clicked", () => {
    const { container } = render(
      <CountrySelectorContext>
        <Tester />
      </CountrySelectorContext>
    );
    screen.getByText("US");
    fireEvent.click(screen.getByText("United States"));
    expect(container).toMatchSnapshot();
    expect(() => {
      screen.getByText("US");
    }).toThrow();
  });
});

describe("coutryString", () => {
  it("should format a list of countries", () => {
    expect(countryString(countries, [])).toEqual("none");
    expect(countryString(countries, ["US"])).toEqual("United States");
    expect(countryString(countries, ["US", "GB"])).toEqual(
      "United Kingdom & United States"
    );
    expect(countryString(countries, ["US", "GB", "CA"])).toEqual(
      "Canada, United Kingdom & United States"
    );
  });
});

describe("sortCountriesByName", () => {
  it("should sort countries by alphabetical order", () => {
    expect(sortCountriesByName(countries)).toEqual([
      countries[2],
      countries[1],
      countries[0],
    ]);
  });
});
