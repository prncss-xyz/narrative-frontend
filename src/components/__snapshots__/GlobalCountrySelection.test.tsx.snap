// Vitest Snapshot v1

exports[`GlobalCountrySelection > should toggle country when clicked 1`] = `
<div>
  <div>
    <div>
      CA
    </div>
    <div>
      GB
    </div>
  </div>
  <div
    class="css-1x3sxzl-Box e2i8k6n3"
  >
    <h3
      class="css-5jl3ge-Box e2i8k6n3"
      color="muted"
      font-size="1"
      font-weight="normal"
      text-decoration="underline"
    >
      Included countries
    </h3>
    <div
      class="css-vq3ac2-Box-Flex e2i8k6n2"
    >
      <div>
        <button
          class="css-1fbado3-Box e2i8k6n3"
          color="text"
          font-size="0"
          height="1"
        >
          Canada
        </button>
      </div>
      <div>
        <button
          class="css-1fbado3-Box e2i8k6n3"
          color="text"
          font-size="0"
          height="1"
        >
          United Kingdom
        </button>
      </div>
      <div>
        <button
          class="css-1gxxcqd-Box e2i8k6n3"
          color="muted"
          font-size="0"
          height="1"
        >
          United States
        </button>
      </div>
    </div>
  </div>
</div>
`;
